src/machine/machine_state_impl.rs

```diff
@@ -558,7 +567,22 @@ impl MachineState {
                                         );
                                     }
                                 }
-                                _ => {
+				(HeapCellValueTag::Cons, ptr) => {
+				    let stream = cell_as_stream!(ptr);
+				    let n2 = stream.options().get_alias().unwrap();
```

Comment:
Similarly to my above comment, `ptr` should be matched to a stream first, and `stream.options().get_alias()` should be destructured in a `match` or `if let` statement.